src/SprykerEco/Zed/Unzer/Business/ApiAdapter/UnzerAbstractApiAdapter.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\Unzer\Business\ApiAdapter;
+
+use Generated\Shared\Transfer\UnzerApiErrorResponseTransfer;
+use Generated\Shared\Transfer\UnzerApiResponseTransfer;
+use SprykerEco\Zed\Unzer\Business\Exception\UnzerApiException;
+
+abstract class UnzerAbstractApiAdapter
+{
+    /**
+     * @param \Generated\Shared\Transfer\UnzerApiResponseTransfer $unzerApiResponseTransfer
+     *
+     * @throws \SprykerEco\Zed\Unzer\Business\Exception\UnzerApiException
+     *
+     * @return void
+     */
+    protected function checkSuccessResponse(UnzerApiResponseTransfer $unzerApiResponseTransfer)
+    {
+        if ($unzerApiResponseTransfer->getIsSuccess()) {
+            return;
+        }
+
+        $errorMessage = $unzerApiResponseTransfer->getErrorResponseOrFail();
+
+        throw new UnzerApiException($this->concatErrors($errorMessage));
```

Comment:
```suggestion
        if (!$unzerApiResponseTransfer->getIsSuccess()) {
            $errorMessage = $unzerApiResponseTransfer->getErrorResponseOrFail();

            throw new UnzerApiException($this->concatErrors($errorMessage));
        }
```

Reply:
I prefer to use early-return statement :)